### ISSUE TITLE
Service had to be Manually Credentialed as Admin to Start

### ISSUE THREAD

**Issue by SixSpeedDriver** [bruhs/caddy-service#20]
Total n00b here, so be gentle. Had a tough enough time to get caddy to start (every guide on Windows said use .txt for the caddyfile, and that doesn't work when you want it to start without parameterizing the config location...but I digress).

Following your guides (and for reference, I was using the command prompt as administrator), and got the service to install, but not start through the CLI.  When i went in through the GUI and told the service to run as Administrator and provided the credentials it started.  Of course, running this as a lower privileged user would be preferred - Is this an issue in my configuration of caddy or in the caddy-service that I had to privilege the service accordingly?

Any help's appreciated!

Side question (further exposing my n00b nature, sorry - I assume there's a good reason for this) - in the readme,md, is there a reason you don't specifically call out the -conf parameter, even though it's required for install?

Would it be better to potentially read as follows?: 

Install a Caddy service:
caddy -service install -conf pathToConfiguration [-name optionalServiceName] [-option optionValue]

I mean, technically it seems like -conf is an optional parameter, but should it be?  Or would making it a required parameter mean you have to provide it for every reference to -service?

**Comment by tobya** [comment]
hi @SixSpeedDriver .  Thanks for the issue.   

Could you link to those Windows that you are talking about?

I agree with you about making  -conf a defacto required parameter, that's a great idea.

On your issue, can you provide the exact command string you used to run your instance of caddy and any errors received.   Please make sure you uses -log to ensure you have a system log.

**Comment by fabiofcferreira** [comment]
Hi @SixSpeedDriver!

Since @tobya has already asked for the logs I think it's not necessary to repeat the request. As soon as you reply, we will be able to help you way faster.

I'm replying due to your side question. We haven't added the `-conf` parameter to the README.md although it is recommended due to the fact that most of the people might not have the Caddyfile in the same directory as the Caddy executable.

All in all, there is no reason for that and we actually never noticed that we missed it.
Thanks for helping us improve our plugin.

**Comment by fabiofcferreira** [comment]
Hi @SixSpeedDriver !

I'm writing this in order to answer you side question. We haven't added `-conf` parameter to the example but it's in the _Notes_ section, and is actually clear that it is required.
Even though, we have already added it to the _Notes_, I am adding it to the example as a required parameter.

Thanks for helping us improve our documentation and for using our plugin.

**Comment by SixSpeedDriver** [comment]
@tobya one example: https://drive.google.com/file/d/0BzPd75yQAJYhbWtRV2JRcnhfbjA
To be clear, that was about the caddy server itself, not this plug-in (so i'm a bit off topic here, was just illustrating my n00bness!).  My first foray into caddy was some time ago and I was using other guides, and kept running into the same problem.  Probably what i get for not firing up an ubuntu VM again, but my linux-fu is pretty garbage.  Something i'm working on.

Anyway, to the meat of the issue - i've run caddy -service stop && a caddy -service uninstall to get fresh again.

I'm installing with (This is on a Windows Server 2012R2 box):
D:\Caddy>caddy -service install -conf="D:\Caddy\caddyfile" -log="D:\Caddy\LogFiles\ServiceRun\ServiceRun.log"

Then, i call  a caddy -service start

No logs generated at that location, looking at services.msc in Windows shows the service didn't start.  
Tried the GUI start for the service, didn't start - mentioned an error to review the System Event logs, this is the XML blob it spit out:

The Caddy service terminated with the following service-specific error: 
Incorrect function.
Event Xml:

```xml
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Service Control Manager" Guid="{555908d1-a6d7-4695-8e1e-26931d2012f4}" EventSourceName="Service Control Manager" />
    <EventID Qualifiers="49152">7024</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>0</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8080000000000000</Keywords>
    <TimeCreated SystemTime="2018-04-16T05:27:35.058076000Z" />
    <EventRecordID>111166</EventRecordID>
    <Correlation />
    <Execution ProcessID="544" ThreadID="9060" />
    <Channel>System</Channel>
    <Computer>Server</Computer>
    <Security />
  </System>
  <EventData>
    <Data Name="param1">Caddy</Data>
    <Data Name="param2">%%1</Data>
    <Binary>430061006400640079000000</Binary>
  </EventData>
</Event>
```

I then went in to the services.msc GUI, set the log on info to Administrator manually, and the service fired right up.
